packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -155,6 +155,19 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
           Cypress.backend('ready:for:domain', { originPolicy: location.originPolicy })
 
           if (err) {
+            if (err?.name === 'ReferenceError') {
+              const wrappedErr = $errUtils.errByPath('switchToDomain.ran_domain_fn_reference_error', {
+                error: err.message,
+              })
+
+              wrappedErr.name = err.name
```

Comment:
This carries forward the `DataCloneError` instead of re-titling it a `CypressError`